bson/src/main/org/bson/codecs/pojo/PojoCodecProvider.java

```diff
@@ -72,6 +74,17 @@ public static Builder builder() {
                 classModel = createClassModel(clazz, conventions);
             }
             return new PojoCodecImpl<T>(classModel, registry, discriminatorLookup);
+        } else if (automatic) {
+            try {
+                classModel = createClassModel(clazz, conventions);
+            } catch (IllegalStateException e) {
+                return null;
+            }
+            if (!clazz.isInterface() && classModel.getPropertyModels().isEmpty()) {
```

Comment:
The example is in the `ConcreteAndNestedAbstractInterfaceModel` class, it has a field that is of type `InterfaceBasedModel` - which we want to be able to handle.